components/engine_rocks/src/event_listener.rs

```diff
@@ -3,23 +3,29 @@
 use crate::rocks_metrics::*;
 
 use file_system::{get_io_type, set_io_type, IOType};
+use regex::Regex;
 use rocksdb::{
     CompactionJobInfo, DBBackgroundErrorReason, FlushJobInfo, IngestionInfo, MutableStatus,
     SubcompactionJobInfo, WriteStallInfo,
 };
+use tikv_util::metrics::CRITICAL_ERROR;
 use tikv_util::set_panic_mark;
+use tikv_util::worker::Scheduler;
+use tikv_util::{error, warn};
 
 // Message for RocksDB status subcode kNoSpace.
 const NO_SPACE_ERROR: &str = "IO error: No space left on device";
 
 pub struct RocksEventListener {
     db_name: String,
+    sst_recovery_tx: Option<Scheduler<String>>,
```

Comment:
```suggestion
    sst_recovery_scheduler: Option<Scheduler<String>>,
```

Reply:
And why exactly are we doing the recovery in another thread?

Reply:
rocksdb event listener thread should not be blocked.